### PR TITLE
Work around React Hot Loader proxying route classes

### PR DESCRIPTION
**Pull request by taion**
cc @itajaja @hollandThomas

This resolves a wonky interaction with React Hot Loader, which was the cause of the issue at https://github.com/4Catalyzer/found/issues/73#issuecomment-339484187.

### CODE REVIEW

**Comment by taion** [comment]
Oh, hmm, we can't get the original component this way.

**Comment by taion** [comment]
It would given the proxy... but we have the proxied component here, not the proxy itself. Annoying.

**Comment by taion** [comment]
Updated and fixed. @itajaja can you please test?

File: README.md [hunk 1]
```diff
@@ -505,6 +505,27 @@ const route = {
 };
 ```
 
+#### Custom route classes
+
+You can implement reusable logic in routes with a custom route class. When extending `Route`, methods defined on the class will be overridden by explicitly specified route properties. You can use custom route classes for either object route configurations or JSX route configurations.
+
+> **Note:** To avoid issues with [React Hot Loader](http://gaearon.github.io/react-hot-loader/), custom route classes should usually extend `Route`.
```
Comment (itajaja) [inline README.md]: shouldn't this always be the recommendation regardless?

**Comment by taion** [comment]
Released: https://github.com/4Catalyzer/found/releases/tag/v0.3.5

**Pull request merged**